pkg/kore/setup.go

```
@@ -384,22 +393,29 @@ func (h hubImpl) ensureTeamsAndClustersIdentified(ctx context.Context) error {
 			updated := false
 			logger := teamlog.WithField("cluster", cluster.Name)
 
+			if cluster.Labels == nil {
+				cluster.Labels = map[string]string{}
+			}
+
+			if cluster.Labels[LabelKoreIdentifier] == "" {
+				logger.Info("setting kore identifier for cluster")
+				cluster.Labels[LabelKoreIdentifier] = h.KoreIdentifier()
+				updated = true
+			}
+
 			if cluster.Labels[LabelTeamIdentifier] == "" {
 				logger.Info("setting team identifier for cluster")
-				if cluster.Labels == nil {
-					cluster.Labels = map[string]string{}
-				}
 				cluster.Labels[LabelTeamIdentifier] = team.Labels[LabelTeamIdentifier]
 				updated = true
 			}
 
 			if cluster.Labels[LabelClusterIdentifier] == "" {
 				logger.Info("assigning cluster identifier")
-
-				if cluster.Labels == nil {
-					cluster.Labels = map[string]string{}
+				cloud, mapErr := h.Costs().Metadata().MapProviderToCloud(cluster.Spec.Kind)
+				if mapErr != nil {
+					log.WithField("provider", cluster.Spec.Kind).WithError(mapErr).Warn("received error mapping k8s provider to cloud provider for asset identifier, will continue with blank cloud provider")
```

we shouldn't swallow this error either.

---

there's at least one known case where this will be nil - the kore cluster doesn't have a cloud. I think in the setup.go, we shouldn't explode if this isn't available, the warning should be enough.

---

We could still map it to something internal, e.g. "kore", "unmanaged". Nobody will ever see that warning

---

fixed in a3336ce